Knoema.Client/Client.cs

```diff
@@ -117,12 +117,18 @@ private Task<T> ApiPost<T>(string path, object obj)
 
 		public Task<IEnumerable<Dataset>> ListDatasets(string source = null, string topic = null, string region = null)
 		{
-			var query = string.Empty
-				.AddUrlParam("source", source)
-				.AddUrlParam("topic", topic)
-				.AddUrlParam("region", region);
 
-			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
+			if (string.IsNullOrEmpty(source ?? topic ?? region))
+				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
+			else
+			{
+				var payload = new Dictionary<string, string>();
```

Comment:
public Task<IEnumerable<Dataset>> Datasets(object payload = null)
{
if (payload == null)
return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
return ApiPost<IEnumerable<Dataset>>("/api/1.0/meta/dataset", payload);
}

// Examples of using
var datasets = await client.Datasets(new { topic = "Argentina" });
var datasets = await client.Datasets(new { topic = "Argentina", source =
"OECD" });
var datasets = await client.Datasets(new { topic = "Electricity", source =
"EIA", region = "AR" });

On Fri, Aug 21, 2015 at 4:45 PM, AlexPhilippov <notifications@github.com>
wrote:

> In Knoema.Client/Client.cs
> <https://github.com/Knoema/knoema-csharp-driver/pull/15#discussion_r37626956>
> :
>
> >
> > -			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
> > +			if (string.IsNullOrEmpty(source ?? topic ?? region))
> > +				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
> > +			else
> > +			{
> > +				var payload = new Dictionary<string, string>();
>
> Probably you could use new anonymous object instead of creating dictionary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Knoema/knoema-csharp-driver/pull/15/files#r37626956>.
>



-- 
Evgeniya Polyakova,
Knoema